### ISSUE TITLE
Syncthing is not connecting anymore

### ISSUE THREAD

**Issue by ghost** [SynoCommunity/spksrc#4138]
After updating Syncthing from 1.6.1-20 to 1.7.1-20 my two Diskstations are permanently disconnected from each other. Both were setup for dynamic address. There was no change on both routers involved. Unfortunately at the same time at least on one of the machines DSM was updated from 6.2.3-25462 update 1 to update 2 so I'm not really sure if the DSM update or the Syncthing update is the cause of the trouble. Meanwhile both are running update 2 but that doesn't help. One of the DS syncs also with a Win10 machine and this works (with dynamic address).

### Setup
_Package Name:_ Syncthing
_Package Version:_ 1.7.1-20

_NAS Model:_ DS213air and DS213J
_NAS Architecture:_ armv5 and armada370
_DSM version:_ 6.2.3-25462 update 1 to update 2

### Expected behavior
Both DS should connect to each other.

### Actual behavior
They don't connect. I tried also setting the IP manually (and opening TCP port 22000 on one router) but that resulted in an "address already in use" error message.

### Steps to reproduce
_1._ 
_2._ 
_3._ 

#### Package log
_Check Package Center or `/usr/local/{package}/var/`_
```
That's odd, I cannot cd into this folder even as admin ...
```
#### Other logs
_E.g. `/var/log/messages` or `/var/log/synopkg.log`_
```
2020/08/17 23:24:52     upgrade syncthing: begin to stop version 1.6.1-20
2020/08/17 23:24:54     upgrade syncthing 1.6.1-20 Begin start-stop-status stop
2020/08/17 23:24:56     upgrade syncthing 1.6.1-20 End start-stop-status stop ret=[0]
2020/08/17 23:24:56     upgrade syncthing 1.6.1-20 Begin unload apparmor
2020/08/17 23:24:56     upgrade syncthing 1.6.1-20 End unload apparmor ret=[0]
2020/08/17 23:24:56     upgrade syncthing: stop version 1.6.1-20 successfully, result 0
2020/08/17 23:24:56     upgrade syncthing 1.6.1-20 Begin preupgrade
2020/08/17 23:25:06     upgrade syncthing 1.6.1-20 End preupgrade ret=[0]
2020/08/17 23:25:06     upgrade syncthing 1.6.1-20 Begin preuninst
2020/08/17 23:25:06     upgrade syncthing 1.6.1-20 End preuninst ret=[0]
2020/08/17 23:25:06     upgrade syncthing 1.6.1-20 Begin /bin/rm -rf /volume1/@appstore/syncthing
2020/08/17 23:25:07     upgrade syncthing 1.6.1-20 End /bin/rm -rf /volume1/@appstore/syncthing ret=[0]
2020/08/17 23:25:07     upgrade syncthing 1.6.1-20 Begin postuninst
2020/08/17 23:25:07     upgrade syncthing 1.6.1-20 End postuninst ret=[0]
2020/08/17 23:25:07     upgrade syncthing: Uninstall 1.6.1-20 successfully
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin preinst
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End preinst ret=[0]
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin /bin/rm -rf /volume1/@appstore/syncthing
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End /bin/rm -rf /volume1/@appstore/syncthing ret=[0]
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin /bin/mv -f /volume1/@tmp/pkginstall/package /volume1/@appstore/syncthing
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End /bin/mv -f /volume1/@tmp/pkginstall/package /volume1/@appstore/syncthing ret=[0]
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin /bin/mkdir -p /var/packages/syncthing
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End /bin/mkdir -p /var/packages/syncthing ret=[0]
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin /bin/mv -f /volume1/@tmp/pkginstall/INFO /var/packages/syncthing/INFO
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End /bin/mv -f /volume1/@tmp/pkginstall/INFO /var/packages/syncthing/INFO ret=[0]
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin /bin/rm -rf /var/packages/syncthing/scripts
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End /bin/rm -rf /var/packages/syncthing/scripts ret=[0]
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin /bin/mv -f /volume1/@tmp/pkginstall/scripts /var/packages/syncthing/scripts
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End /bin/mv -f /volume1/@tmp/pkginstall/scripts /var/packages/syncthing/scripts ret=[0]
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin /bin/rm -rf /var/packages/syncthing/WIZARD_UIFILES
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End /bin/rm -rf /var/packages/syncthing/WIZARD_UIFILES ret=[0]
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin /bin/mv -f /volume1/@tmp/pkginstall/WIZARD_UIFILES /var/packages/syncthing/WIZARD_UIFILES
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End /bin/mv -f /volume1/@tmp/pkginstall/WIZARD_UIFILES /var/packages/syncthing/WIZARD_UIFILES ret=[0]
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin /bin/rm -rf /var/packages/syncthing/conf
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End /bin/rm -rf /var/packages/syncthing/conf ret=[0]
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 Begin /bin/mv -f /volume1/@tmp/pkginstall/conf /var/packages/syncthing/conf
2020/08/17 23:25:08     upgrade syncthing 1.7.1-20 End /bin/mv -f /volume1/@tmp/pkginstall/conf /var/packages/syncthing/conf ret=[0]
2020/08/17 23:25:10     upgrade syncthing 1.7.1-20 Begin postinst
2020/08/17 23:25:12     upgrade syncthing 1.7.1-20 End postinst ret=[0]
2020/08/17 23:25:12     upgrade syncthing 1.7.1-20 Begin postupgrade
2020/08/17 23:25:19     upgrade syncthing 1.7.1-20 End postupgrade ret=[0]
2020/08/17 23:25:23     upgrade syncthing: begin to start version 1.7.1-20
2020/08/17 23:25:23     upgrade syncthing 1.7.1-20 Begin pre-load apparmor
2020/08/17 23:25:24     upgrade syncthing 1.7.1-20 End pre-load apparmor ret=[0]
2020/08/17 23:25:24     upgrade syncthing 1.7.1-20 Begin start-stop-status start
2020/08/17 23:25:24     upgrade syncthing 1.7.1-20 End start-stop-status start ret=[0]
2020/08/17 23:25:25     upgrade syncthing: start version 1.7.1-20 successfully, result 0
```

**Comment by acolomb** [comment]
Was that a test build of version v1.6.1-20? Because Such an SPK package version was never published and only existed as test build in a PR.

I don't have any conrete ideas what your issue might be, but it doesn't really sound like a Synology specific problem. If you have already checked the firewall settings...

Can you access Syncthing's web GUI on both machines? What does it say there? Especially check the "Listeners" in your own device status section and maybe look into the messages under "Actions" -> "Logs" for hints about the connection failures.

**Comment by juergent20** [comment]
You generated it for me because of issue https://github.com/SynoCommunity/spksrc/issues/4026

I would would also think it's a more generic issue if it not happened right after an update.

When I click on Disconery under This Device I get currently the following on both machines:

global@https://discovery-v4.syncthing.net/v2/: Post "https://discovery-v4.syncthing.net/v2/": dial tcp 0.0.0.0:22000->139.59.84.212:443: bind: address already in use
global@https://discovery-v6.syncthing.net/v2/: Post "https://discovery-v6.syncthing.net/v2/": dial tcp 0.0.0.0:22000->[2400:6180:100:d0::741:a001]:443: bind: address already in use

On the machine which connects successfully to the PC but not to the DS I see this lines in the log:

2020-08-22 10:34:46 Listen (BEP/relay): Get "https://relays.syncthing.net/endpoint": net/http: TLS handshake timeout
2020-08-22 10:34:46 Relay listener (dynamic+https://relays.syncthing.net/endpoint) shutting down
2020-08-22 10:34:46 c.S.listenerSupervisor: Failed service 'dynamic+https://relays.syncthing.net/endpoint' (1.000000 failures of 2.000000), restarting: true, error: "{dynamic+https://relays.syncthing.net/endpoint dynamic+https://relays.syncthing.net/endpoint} returned unexpectedly", stacktrace: [unknown stack trace]

On the the other DS it's only

2020-08-24 08:34:12 My ID: ...
2020-08-24 08:34:18 Single thread SHA256 performance is 5.3 MB/s using crypto/sha256 (1.4 MB/s using minio/sha256-simd).
2020-08-24 08:34:38 Hashing performance is 1.11 MB/s
2020-08-24 08:34:38 Overall send rate is unlimited, receive rate is unlimited
2020-08-24 08:34:38 Using discovery server https://discovery.syncthing.net/v2/?noannounce&id=LYXKCHX-VI3NYZR-ALCJBHF-WMZYSPK-QG6QJA3-MPFYMSO-U56GTUK-NA2MIAW
2020-08-24 08:34:38 Using discovery server https://discovery-v4.syncthing.net/v2/?nolookup&id=LYXKCHX-VI3NYZR-ALCJBHF-WMZYSPK-QG6QJA3-MPFYMSO-U56GTUK-NA2MIAW
2020-08-24 08:34:38 Using discovery server https://discovery-v6.syncthing.net/v2/?nolookup&id=LYXKCHX-VI3NYZR-ALCJBHF-WMZYSPK-QG6QJA3-MPFYMSO-U56GTUK-NA2MIAW
2020-08-24 08:34:39 Relay listener (dynamic+https://relays.syncthing.net/endpoint) starting
2020-08-24 08:34:39 Ready to synchronize "..." (...) (sendreceive)
2020-08-24 08:34:39 QUIC listener ([::]:22000) starting
2020-08-24 08:34:39 ...
2020-08-24 08:49:22 Relay listener (dynamic+https://relays.syncthing.net/endpoint) starting
2020-08-24 08:49:57 Joined relay relay://159.69.201.90:22067
2020-08-24 09:03:52 Failed to exchange Hello messages with ... at [::]:22000-130.183.250.2:48833/quic-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: Application error 0x0: closing
2020-08-24 09:09:12 Failed to exchange Hello messages with ... at [::]:22000-130.183.250.2:31596/quic-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: EOF
2020-08-24 09:15:54 Failed to exchange Hello messages with ... at [::]:22000-130.183.250.2:43454/quic-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: EOF
2020-08-24 09:24:53 Failed to exchange Hello messages with ... at [::]:22000-130.183.250.2:48604/quic-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: Application error 0x0: closing
2020-08-24 09:40:34 Failed to exchange Hello messages with ... at [::]:22000-130.183.250.2:32237/quic-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: EOF
2020-08-24 10:26:34 quic://0.0.0.0:22000 resolved external address quic://93.222.83.248:22100 (via stun.syncthing.net:3478)
2020-08-24 11:40:15 Failed to exchange Hello messages with ... at 192.168.2.100:22000-130.183.250.2:7186/tcp-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: write tcp 192.168.2.100:22000->130.183.250.2:7186: write: connection reset by peer
2020-08-24 11:40:50 Pausing 2H6RVOP-T3TERPY-R57YCKF-EK47TJP-4CQRYLJ-EEAIAH3-23DW4WJ-GV6DOQA
2020-08-24 11:41:25 Failed to exchange Hello messages with ... at 192.168.2.100:22000-130.183.250.2:16736/tcp-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: write tcp 192.168.2.100:22000->130.183.250.2:16736: write: connection reset by peer
2020-08-24 11:42:35 Failed to exchange Hello messages with ... at 192.168.2.100:22000-130.183.250.2:57737/tcp-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: write tcp 192.168.2.100:22000->130.183.250.2:57737: write: connection reset by peer
2020-08-24 11:43:45 Failed to exchange Hello messages with ... at 192.168.2.100:22000-130.183.250.2:50876/tcp-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: write tcp 192.168.2.100:22000->130.183.250.2:50876: write: connection reset by peer

**Comment by acolomb** [comment]
>     When I click on Disconery under This Device I get currently the following on both machines:

I meant the "Listeners" heading, just above "Discovery". On my machine it says 3/3 for example, meaning that it has three external points for connection to this instance. [By default](https://docs.syncthing.net/users/config.html#listen-addresses), they are configured as TCP, QUIC and relays, and the status shows how many of these mechanisms are actually functional.  If they all failed, no connections from the outside are possible.

>     global@https://discovery-v4.syncthing.net/v2/: Post "https://discovery-v4.syncthing.net/v2/": dial tcp 0.0.0.0:22000->139.59.84.212:443: bind: address already in use
>     global@https://discovery-v6.syncthing.net/v2/: Post "https://discovery-v6.syncthing.net/v2/": dial tcp 0.0.0.0:22000->[2400:6180100d0::741:a001]:443: bind: address already in use

That sounds strange already, but I'm not quite sure where you're seeing this, as I can't "click" on Discovery? The `bind: address already in use` part hints to another instance of Syncthing already running on this machine. But the connections to the discovery servers are outgoing, so it's unclear why it would try to bind to a network interface local port.

>     On the machine which connects successfully to the PC but not to the DS I see this lines in the log:
>     
>     2020-08-22 10:34:46 Listen (BEP/relay): Get "https://relays.syncthing.net/endpoint": net/http: TLS handshake timeout
>     2020-08-22 10:34:46 Relay listener (dynamic+https://relays.syncthing.net/endpoint) shutting down
>     2020-08-22 10:34:46 c.S.listenerSupervisor: Failed service 'dynamic+https://relays.syncthing.net/endpoint' (1.000000 failures of 2.000000), restarting: true, error: "{dynamic+https://relays.syncthing.net/endpoint dynamic+[https://relays.syncthing.net/endpoint}](https://relays.syncthing.net/endpoint%7D) returned unexpectedly", stacktrace: [unknown stack trace]

This means that Syncthing cannot connect to the relay network. Most likely something is blocking internet access from this machine to the outside.  Since it is regular HTTPS traffic, that problem would affect other software on your DS as well.

The relay network is only a fall-back connection, you should strive toward direct connections among all devices.

>     On the the other DS it's only
>     
>     2020-08-24 08:34:12 My ID: ...
>     2020-08-24 08:34:18 Single thread SHA256 performance is 5.3 MB/s using crypto/sha256 (1.4 MB/s using minio/sha256-simd).
>     2020-08-24 08:34:38 Hashing performance is 1.11 MB/s
>     2020-08-24 08:34:38 Overall send rate is unlimited, receive rate is unlimited
>     2020-08-24 08:34:38 Using discovery server https://discovery.syncthing.net/v2/?noannounce&id=LYXKCHX-VI3NYZR-ALCJBHF-WMZYSPK-QG6QJA3-MPFYMSO-U56GTUK-NA2MIAW
>     2020-08-24 08:34:38 Using discovery server https://discovery-v4.syncthing.net/v2/?nolookup&id=LYXKCHX-VI3NYZR-ALCJBHF-WMZYSPK-QG6QJA3-MPFYMSO-U56GTUK-NA2MIAW
>     2020-08-24 08:34:38 Using discovery server https://discovery-v6.syncthing.net/v2/?nolookup&id=LYXKCHX-VI3NYZR-ALCJBHF-WMZYSPK-QG6QJA3-MPFYMSO-U56GTUK-NA2MIAW
>     2020-08-24 08:34:39 Relay listener (dynamic+https://relays.syncthing.net/endpoint) starting
>     2020-08-24 08:34:39 Ready to synchronize "..." (...) (sendreceive)
>     2020-08-24 08:34:39 QUIC listener ([::]:22000) starting
>     2020-08-24 08:34:39 ...
>     2020-08-24 08:49:22 Relay listener (dynamic+https://relays.syncthing.net/endpoint) starting
>     2020-08-24 08:49:57 Joined relay relay://159.69.201.90:22067
>     2020-08-24 09:03:52 Failed to exchange Hello messages with ... at [::]:22000-130.183.250.2:48833/quic-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: Application error 0x0: closing
>     2020-08-24 09:09:12 Failed to exchange Hello messages with ... at [::]:22000-130.183.250.2:31596/quic-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: EOF
>     2020-08-24 09:15:54 Failed to exchange Hello messages with ... at [::]:22000-130.183.250.2:43454/quic-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: EOF
>     2020-08-24 09:24:53 Failed to exchange Hello messages with ... at [::]:22000-130.183.250.2:48604/quic-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: Application error 0x0: closing
>     2020-08-24 09:40:34 Failed to exchange Hello messages with ... at [::]:22000-130.183.250.2:32237/quic-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: EOF
>     2020-08-24 10:26:34 quic://0.0.0.0:22000 resolved external address quic://93.222.83.248:22100 (via stun.syncthing.net:3478)
>     2020-08-24 11:40:15 Failed to exchange Hello messages with ... at 192.168.2.100:22000-130.183.250.2:7186/tcp-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: write tcp 192.168.2.100:22000->130.183.250.2:7186: write: connection reset by peer
>     2020-08-24 11:40:50 Pausing 2H6RVOP-T3TERPY-R57YCKF-EK47TJP-4CQRYLJ-EEAIAH3-23DW4WJ-GV6DOQA
>     2020-08-24 11:41:25 Failed to exchange Hello messages with ... at 192.168.2.100:22000-130.183.250.2:16736/tcp-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: write tcp 192.168.2.100:22000->130.183.250.2:16736: write: connection reset by peer
>     2020-08-24 11:42:35 Failed to exchange Hello messages with ... at 192.168.2.100:22000-130.183.250.2:57737/tcp-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: write tcp 192.168.2.100:22000->130.183.250.2:57737: write: connection reset by peer
>     2020-08-24 11:43:45 Failed to exchange Hello messages with ... at 192.168.2.100:22000-130.183.250.2:50876/tcp-server/TLS1.3-TLS_CHACHA20_POLY1305_SHA256: write tcp 192.168.2.100:22000->130.183.250.2:50876: write: connection reset by peer

Here the relay and discovery servers are both successfully connected. The `connection reset by peer` messages sound scary, but are actually pretty normal if two devices connect to each other (directly, in both directions) and then one of the redundant connections is dropped. It might point to a network problem though. You should watch out for the involved IP addresses and try to correlate if a connection attempt on one side even appears in the log at the other side.

All in all I think you could probably get more help and support in the [Syncthing forum](https://forum.syncthing.net/), since it seems to be unspecific to this DSM package.

**Comment by juergent20** [comment]
Thank you very much for explaining all this. On the Listeners I see 3/3. On discovery there is 3/5, this is probably why I can click on it to get failure information.

I reinstalled Syncthing on both machine but that didN#t help.

netstat on one of the DS gives me

tcp6       0      0 [::]:22000              [::]:*                  LISTEN
udp6       0      0 [::]:22000              [::]:*

After uninstalling Syncthing both entries are gone.

As you suggested I'll contact the Syncthing forum.

Again, thanks!

**Comment by juergent20** [comment]
After several attempts to reinstall, open ports on the routers and so on, it finally works again. It seems the auto update to v1.9.0 solved the issue.

**Comment by acolomb** [comment]
Then it might be related to https://github.com/syncthing/syncthing/issues/6892 which is mentioned in the change log for v1.9.0. Just a guess, but might explain the self healing.

Happy it's working for you again.

**Comment by hgy59** [comment]
fixed since syncthing v1.9.0
